tests/integration/suite/test_app.py

```diff
@@ -760,6 +760,26 @@ def test_app_externalid_target_project_verification(admin_mc,
     p1_client.update(app, update_data)
 
 
+def test_local_app_can_deploy(admin_pc, admin_mc, remove_resource):
+    """Test that an app without an externalId can be deployed
+    successfully to simulate a local app deployed through cli"""
+    app_client = admin_pc.client
+    app_name = random_str()
+    ns = admin_pc.cluster.client.create_namespace(name=random_str(),
+                                                  projectId=admin_pc.
+                                                  project.id)
+    remove_resource(ns)
+    # create app without the externalId value set
+    app = app_client.create_app(
+        name=app_name,
+        targetNamespace=ns.name,
+        projectId=admin_pc.project.id,
+    )
+    remove_resource(app)
+    # check that app is listed as an available app
+    assert app_client.list_app(id=app.id)
```

Comment:
The only other places I see list used this way, it uses name instead of id. I think even if it doesn't exist, a list will not fail, you would need to assert the length or maybe that it isn't none. It may be easier to do a call to by_id_app.